prov/efa/src/efa_mr.c

```diff
@@ -80,6 +88,7 @@ int efa_mr_cache_entry_reg(struct ofi_mr_cache *cache,
 
 	md->mr_fid.mem_desc = (void *)(uintptr_t)md->mr->lkey;
 	md->mr_fid.key = md->mr->rkey;
+	md->is_new_entry_reg = true;
```

Comment:
Why not just call efa_mr_shm_reg() and efa_mr_map_insert() here?

Reply:
For efa_mr_map_insert and efa_mr_shm_reg, we require attr object which is available in efa_mr_cache_reg, hence the boolean.